src/components/Semester/Semester.vue

```diff
@@ -347,6 +350,19 @@ export default defineComponent({
           ),
         })
       );
+      this.openConfirmationModal(`Changed color for ${code}`);
+    },
+    colorSubject(color: string, targetCourse: string) {
+      const code = targetCourse.split(' ')[0];
```

Comment:
Can you rename either this variable or the one inside the `colorCourse` function? I don't even know which one is right, but it's confusing that they're both `code` since I know we're looking at 2 different properties of a course right now.